pkg/sdk/procedures_impl_gen.go

```diff
@@ -0,0 +1,579 @@
+		opts.Returns = &ProcedureReturns2{
+			ResultDataType: r.Returns.ResultDataType,
+			NotNull:        r.Returns.NotNull,
+		}
+	}
+	if r.StrictOrNot != nil {
+		opts.StrictOrNot = &ProcedureStrictOrNot{
+			Strict:            r.StrictOrNot.Strict,
+			CalledOnNullInput: r.StrictOrNot.CalledOnNullInput,
+		}
+	}
+	if r.VolatileOrNot != nil {
+		opts.VolatileOrNot = &ProcedureVolatileOrNot{
+			Volatile:  r.VolatileOrNot.Volatile,
+			Immutable: r.VolatileOrNot.Immutable,
+		}
+	}
+	if r.ExecuteAs != nil {
+		opts.ExecuteAs = &ProcedureExecuteAs{
+			Caller: r.ExecuteAs.Caller,
+			Owner:  r.ExecuteAs.Owner,
+		}
+	}
+	return opts
+}
+
+func (r *CreateProcedureForPythonProcedureRequest) toOpts() *CreateProcedureForPythonProcedureOptions {
+	opts := &CreateProcedureForPythonProcedureOptions{
+		OrReplace: r.OrReplace,
+		Secure:    r.Secure,
+		name:      r.name,
+
+		CopyGrants: r.CopyGrants,
+
+		RuntimeVersion: r.RuntimeVersion,
+
+		Handler:                    r.Handler,
+		ExternalAccessIntegrations: r.ExternalAccessIntegrations,
+
+		Comment: r.Comment,
+
+		As: r.As,
+	}
+	if r.Arguments != nil {
+		s := make([]ProcedureArgument, len(r.Arguments))
+		for i, v := range r.Arguments {
+			s[i] = ProcedureArgument{
+				ArgName:     v.ArgName,
+				ArgDataType: v.ArgDataType,
+			}
+		}
+		opts.Arguments = s
+	}
+	if r.Returns != nil {
+		opts.Returns = &ProcedureReturns{}
+		if r.Returns.ResultDataType != nil {
+			opts.Returns.ResultDataType = &ProcedureReturnsResultDataType{
+				ResultDataType: r.Returns.ResultDataType.ResultDataType,
+				Null:           r.Returns.ResultDataType.Null,
+				NotNull:        r.Returns.ResultDataType.NotNull,
+			}
+		}
+		if r.Returns.Table != nil {
+			opts.Returns.Table = &ProcedureReturnsTable{}
+			if r.Returns.Table.Columns != nil {
+				s := make([]ProcedureColumn, len(r.Returns.Table.Columns))
+				for i, v := range r.Returns.Table.Columns {
+					s[i] = ProcedureColumn{
+						ColumnName:     v.ColumnName,
+						ColumnDataType: v.ColumnDataType,
+					}
+				}
+				opts.Returns.Table.Columns = s
+			}
+		}
+	}
+	if r.Packages != nil {
+		s := make([]ProcedurePackage, len(r.Packages))
+		for i, v := range r.Packages {
+			s[i] = ProcedurePackage{
+				Package: v.Package,
+			}
+		}
+		opts.Packages = s
+	}
+	if r.Imports != nil {
+		s := make([]ProcedureImport, len(r.Imports))
+		for i, v := range r.Imports {
+			s[i] = ProcedureImport{
+				Import: v.Import,
+			}
+		}
+		opts.Imports = s
+	}
+	if r.Secrets != nil {
+		s := make([]ProcedureSecret, len(r.Secrets))
+		for i, v := range r.Secrets {
+			s[i] = ProcedureSecret{
+				SecretVariableName: v.SecretVariableName,
+				SecretName:         v.SecretName,
+			}
+		}
+		opts.Secrets = s
+	}
+	if r.StrictOrNot != nil {
+		opts.StrictOrNot = &ProcedureStrictOrNot{
+			Strict:            r.StrictOrNot.Strict,
+			CalledOnNullInput: r.StrictOrNot.CalledOnNullInput,
+		}
+	}
+	if r.VolatileOrNot != nil {
+		opts.VolatileOrNot = &ProcedureVolatileOrNot{
```

Comment:
#nit-picking
this part (and some others) are repeated for multiple methods, we could extract the common behaviour for them to make the convert code less lengthy